executors/ethereum/migrationBatchCreator.go

```diff
@@ -0,0 +1,134 @@
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+)
+
+var zero = big.NewInt(0)
+
+// ArgsMigrationBatchCreator is the argument for the NewMigrationBatchCreator constructor
+type ArgsMigrationBatchCreator struct {
+	TokensList           []string
+	TokensMapper         TokensMapper
+	Erc20ContractsHolder Erc20ContractsHolder
+	SafeContractAddress  common.Address
+	SafeContractWrapper  SafeContractWrapper
+}
+
+type migrationBatchCreator struct {
+	tokensList           []string
+	tokensMapper         TokensMapper
+	erc20ContractsHolder Erc20ContractsHolder
+	safeContractAddress  common.Address
+	safeContractWrapper  SafeContractWrapper
+}
+
+// NewMigrationBatchCreator creates a new instance of type migrationBatchCreator that is able to generate the migration batch output file
+func NewMigrationBatchCreator(args ArgsMigrationBatchCreator) (*migrationBatchCreator, error) {
+	if len(args.TokensList) == 0 {
+		return nil, errEmptyTokensList
+	}
+	if check.IfNil(args.TokensMapper) {
+		return nil, errNilTokensMapper
+	}
+	if check.IfNil(args.Erc20ContractsHolder) {
+		return nil, errNilErc20ContractsHolder
+	}
+	if check.IfNilReflect(args.SafeContractWrapper) {
+		return nil, errNilSafeContractWrapper
+	}
+
+	return &migrationBatchCreator{
+		tokensList:           args.TokensList,
+		tokensMapper:         args.TokensMapper,
+		erc20ContractsHolder: args.Erc20ContractsHolder,
+		safeContractAddress:  args.SafeContractAddress,
+		safeContractWrapper:  args.SafeContractWrapper,
+	}, nil
+}
+
+// CreateBatchInfo creates an instance of type BatchInfo
+func (creator *migrationBatchCreator) CreateBatchInfo(ctx context.Context, newSafeAddress common.Address) (*BatchInfo, error) {
+	batchesCount, err := creator.safeContractWrapper.BatchesCount(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return nil, err
+	}
+
+	depositsCount, err := creator.safeContractWrapper.DepositsCount(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return nil, err
+	}
+
+	deposits, err := creator.fetchERC20ContractsAddresses(ctx, depositsCount)
+	if err != nil {
+		return nil, err
+	}
+
+	err = creator.fetchBalances(ctx, deposits)
+	if err != nil {
+		return nil, err
+	}
+
+	return creator.assembleBatchInfo(batchesCount, deposits, newSafeAddress)
+}
+
+func (creator *migrationBatchCreator) fetchERC20ContractsAddresses(ctx context.Context, lastDepositNonce uint64) ([]*DepositInfo, error) {
+	deposits := make([]*DepositInfo, 0, len(creator.tokensList))
+	for idx, token := range creator.tokensList {
+		addressBytes, err := creator.tokensMapper.ConvertToken(ctx, []byte(token))
+		if err != nil {
+			return nil, err
+		}
+
+		deposit := &DepositInfo{
+			DepositNonce:    lastDepositNonce + uint64(1+idx),
+			Token:           token,
+			ContractAddress: common.BytesToAddress(addressBytes).String(),
+			contractAddress: common.BytesToAddress(addressBytes),
```

Comment:
I would not have 1 private and 1 public member for a single purpose, but rather have a .String() method